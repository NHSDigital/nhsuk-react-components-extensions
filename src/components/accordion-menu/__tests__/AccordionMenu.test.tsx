import React from 'react';
import {render} from '@testing-library/react'
import AccordionMenu from '..';

describe('AccordionMenu', () => {
  it('matches snapshot', () => {
    const component = render(<AccordionMenu />);
    expect(component).toMatchSnapshot();
  });

  describe('Section', () => {
    it('matches snapshot', () => {
      const component = render(<AccordionMenu.Section heading="TestHeading" />);
      expect(component).toMatchSnapshot();
    });

    it('passes through text', () => {
      const component = render(<AccordionMenu.Section heading="TestHeading" />);
      expect(component.container.textContent).toBe(
        'TestHeading',
      );
    });

    // 3363 - fix this one!
    // it('handles defaultOpen', () => {
    //   const component = render(
    //     <AccordionMenu.Section id="accordion" heading="Heading" defaultOpen />,
    //   );
    //   // const event = {
    //   //   preventDefault: jest.fn(),
    //   // };

    //   console.log(`I am component: ${component.container.querySelector('#accordion')?.textContent}`)
    //   console.log(`I am component 2: ${component.getByText('Heading')}`)
    //   console.log(`I am component 2: ${component.container.querySelector('.nhsuk-accordion-menu__section')?.textContent}`)

    //   expect(component.container.querySelector('#accordion')).toBe(4)

      // component.getByText('Heading').click();
      // component.container.querySelector('.nhsuk-accordion-menu__icon')?.click()
      // expect(event.preventDefault).toHaveBeenCalledTimes(1);

      // expect(component.find('#accordion').getElement().props.open).toBe(true);
      // component.find('summary').simulate('click', event);
      // expect(event.preventDefault).toHaveBeenCalledTimes(1);
      // expect(component.find('#accordion').getElement().props.open).toBe(false);
      // component.find('summary').simulate('click', event);
      // expect(event.preventDefault).toHaveBeenCalledTimes(2);
      // expect(component.find('#accordion').getElement().props.open).toBe(true);
      // component.unmount();
    });

    // it('handles open={true}', () => {
    //   const component = shallow(<AccordionMenu.Section id="accordion" heading="Heading" open />);
    //   const event = {
    //     preventDefault: jest.fn(),
    //   };
    //   expect(component.find('#accordion').getElement().props.open).toBe(true);
    //   component.find('summary').simulate('click', event);
    //   expect(event.preventDefault).toHaveBeenCalledTimes(1);
    //   expect(component.find('#accordion').getElement().props.open).toBe(true);
    //   component.find('summary').simulate('click', event);
    //   expect(event.preventDefault).toHaveBeenCalledTimes(2);
    //   expect(component.find('#accordion').getElement().props.open).toBe(true);
    //   component.unmount();
    // });

    // it('handles open={false}', () => {
    //   const component = shallow(
    //     <AccordionMenu.Section id="accordion" heading="Heading" open={false} />,
    //   );
    //   const event = {
    //     preventDefault: jest.fn(),
    //   };
    //   expect(component.find('#accordion').getElement().props.open).toBe(false);
    //   component.find('summary').simulate('click', event);
    //   expect(event.preventDefault).toHaveBeenCalledTimes(1);
    //   expect(component.find('#accordion').getElement().props.open).toBe(false);
    //   component.find('summary').simulate('click', event);
    //   expect(event.preventDefault).toHaveBeenCalledTimes(2);
    //   expect(component.find('#accordion').getElement().props.open).toBe(false);
    //   component.unmount();
    // });
  });
  describe('Link', () => {
    it('matches snapshot', () => {
      const component = render(<AccordionMenu.Link>Test</AccordionMenu.Link>);
      expect(component).toMatchSnapshot();
      expect(component.container.textContent).toBe('Test');
    });
  });
// });
