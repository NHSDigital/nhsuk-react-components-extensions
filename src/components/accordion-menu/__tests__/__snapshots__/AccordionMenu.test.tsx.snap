// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionMenu Link matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nhsuk-accordion-menu__subsection"
      >
        <a
          class="nhsuk-accordion-menu__subsection-link"
        >
          Test
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nhsuk-accordion-menu__subsection"
    >
      <a
        class="nhsuk-accordion-menu__subsection-link"
      >
        Test
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionMenu Section matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <details
        class="nhsuk-accordion-menu__section"
      >
        <summary
          class="nhsuk-accordion-menu__section-summary"
          role="tab"
          tabindex="1"
        >
          <span
            class="nhsuk-accordion-menu__section-summary-text"
          >
            TestHeading
          </span>
          <svg
            aria-hidden="true"
            class="nhsuk-accordion-menu__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="16"
              cy="16"
              fill="#ffffff"
              r="16"
            />
            <path
              d="M16 8v16M8 16h16"
              fill="none"
              stroke="#003087"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="4"
            />
          </svg>
        </summary>
      </details>
    </div>
  </body>,
  "container": <div>
    <details
      class="nhsuk-accordion-menu__section"
    >
      <summary
        class="nhsuk-accordion-menu__section-summary"
        role="tab"
        tabindex="1"
      >
        <span
          class="nhsuk-accordion-menu__section-summary-text"
        >
          TestHeading
        </span>
        <svg
          aria-hidden="true"
          class="nhsuk-accordion-menu__icon"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            fill="#ffffff"
            r="16"
          />
          <path
            d="M16 8v16M8 16h16"
            fill="none"
            stroke="#003087"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="4"
          />
        </svg>
      </summary>
    </details>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionMenu matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nhsuk-accordion-menu"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="nhsuk-accordion-menu"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
